src/value.rs

```diff
@@ -27,35 +29,144 @@ impl NumericValue {
     }
 }
 
-#[derive(PartialEq, Debug)]
+impl PartialOrd for NumericValue {
```

Comment:
This is my main concern in this PR: I see why we need the trait, but **I don't like to make it public API**. Maybe we should use a (private) function instead.